lib/matplotlib/units.py

```diff
@@ -51,7 +51,7 @@ def default_units(x, axis):
 from matplotlib import cbook
 
 
-class ConversionError(TypeError):
+class ConversionError(ValueError):
```

Comment:
This would be an API change. Not sure it's needed or even desirable. It's raised when `convert_units(x)` cannot process the argument. Typically that should happen rather if x has an unsupported type than value.

Reply:
Looking at where it is used 

https://github.com/matplotlib/matplotlib/blob/c30129ba5ed9fe1e3957a8b9c68699a193ff2257/lib/matplotlib/axis.py#L1466-L1470

I personally would have thought that should have been a `ValueError`, but I can understand why in some cases you can argue one way or another for `ValueError` vs `TypeError`. I do agree this would be a breaking change if someone is expecting to catch a `TypeError` instead of a `ConversionError` and maybe you feel strongly enough that this shouldn't change. This is partially changed because I'm more under the impression it should be a `ValueError`, and I did end up using the exception a couple more places and throwing this instead of a `ValueError`. The appropriate fix may be creating two different errors or maybe just documenting an API change.

So in summary: I did this because it didn't seem like it would be a big deal and I am using the exception more widely than it was used before (new use cases with the new method I added, and for existing errors that were both `ValueError`s and seemed like they should have actually fallen under a `ConversionError`)